tests/php/Extension/FluentSiteTreeExtensionTest.php

```diff
@@ -150,4 +151,97 @@ public function testFluentURLs($domain, $locale, $prefixDisabled, $pageName, $ur
         $page = $this->objFromFixture('Page', $pageName);
         $this->assertEquals($url, $page->Link());
     }
+
+    public function testUpdateStatusFlagsInherited()
+    {
+        /** @var Page|FluentSiteTreeExtension $page */
+        $page = $this->objFromFixture('Page', 'home');
+        $flags = $page->getStatusFlags();
+
+        $this->assertTrue(array_key_exists('fluentinherited', $flags));
+    }
+
+    public function testUpdateStatusFlagsDrafted()
+    {
+        /** @var Page|FluentSiteTreeExtension $page */
+        $page = $this->objFromFixture('Page', 'about');
+        $flags = $page->getStatusFlags();
+
+        $this->assertTrue(array_key_exists('modified', $flags));
+
+        if (!array_key_exists('modified', $flags)) {
+            return;
+        }
+
+        $this->assertEquals('Locale drafted', $flags['modified']['text']);
+    }
+
+    public function testUpdateCMSFields()
+    {
+        /** @var Page|FluentSiteTreeExtension $page */
+        $page = $this->objFromFixture('Page', 'home');
+        $fields = new FieldList();
+
+        $page->updateCMSFields($fields);
+
+        $this->assertNotNull($fields->fieldByName('LocaleStatusMessage'));
+    }
+
+    public function testUpdateCMSActionsInherited()
```

Comment:
`testUpdateCMSActionsInherited()` and `testUpdateCMSActionsDrafted()` are very similar, and I considered breaking a large portion of their shared logic out into a separate method, but, ultimately I felt it was better to have all of the assertions in the test methods themselves, rather than out in a shared method. I'd be happy to change this though.